src/test/java/org/mule/runtime/dsl/internal/xml/parser/MuleDocumentLoaderTestCase.java

```diff
@@ -71,4 +62,61 @@ public void documentLoaderShouldNotUseThreadCurrentClassLoader() throws Exceptio
     assertThat(document, is(notNullValue()));
     assertThat(errorHandler.getErrors().size(), is(0));
   }
+
+  @Test
+  public void xmlMetadataIsProperlyPopulatedWhenParsingSimpleApplication() throws Exception {
+
+    InputStream inputStream = currentThread().getContextClassLoader().getResourceAsStream("simple_application.xml");
+    MuleDocumentLoader loader = new MuleDocumentLoader();
+
+    InputSource is = new InputSource(inputStream);
+    XmlGathererErrorHandler errorHandler = new DefaultXmlGathererErrorHandlerFactory().create();
+    Document document = loader.loadDocument(SAXParserFactory::newInstance, is, null, errorHandler, 0, false, null);
+
+    assertThat(document, is(notNullValue()));
+    assertThat(errorHandler.getErrors().size(), is(0));
+
+    XmlMetadataAnnotations rootAnnotations =
+        (XmlMetadataAnnotations) document.getDocumentElement().getUserData(METADATA_ANNOTATIONS_KEY);
+    assertThat(rootAnnotations, is(not(nullValue())));
+    assertThat(rootAnnotations.isSelfClosing(), is(false));
+    // FIXME: these are currently returning the beginning of the document instead of the beginning of the first tag
```

Comment:
I left this FIXME here so we can start validating the general design while I work on a solution.